File: src/java/magma/app/compile/define/Placeholder.java
package magma.app.compile.define;

import magma.app.compile.node.Node;

public record Placeholder(String input) implements Node {
    public static String generatePlaceholder(String input) {
        var replaced = input
                .replace("/*", "start")
                .replace("*/", "end");

        return "/*" + replaced + "*/";
    }

    public boolean is(String type) {
        return false;
    }
}

File: src/ts/magma/app/compile/define/Placeholder.ts
import { Node } from "../../../../magma/app/compile/node/Node";
export class Placeholder implements Node {
	input: string;
	constructor (input: string) {
		this.input = input;
	}
	static generatePlaceholder(input: string): string {
		let replaced = input.replace("/*", "start").replace("*/", "end")/*unknown*/;
		return "/*" + replaced + "*/"/*unknown*/;
	}
	is(type: string): boolean {
		return false/*unknown*/;
	}
}
